includes/Core/Storage/Setting.php

```diff
@@ -108,6 +110,22 @@ public function delete() {
 		return $this->options->delete( static::OPTION );
 	}
 
+	/**
+	 * Validates the setting.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param mixed $value Setting value. Must be serializable if non-scalar.
+	 * @return mixed|WP_Error Validated setting value, or error object on failure.
+	 */
+	final public function validate( $value ) {
+		$callback = $this->get_validate_callback();
+		if ( $callback ) {
+			return $callback( $value );
+		}
+		return $value;
+	}
```

Comment:
Instead of using a callback for validation, why not allow this method to be overloaded by a sub-class?

This method could then simply return `$value` by default and the current consuming classes would override this instead of the get callback method, which would require less duplication.

E.g.
```php
public function validate( $value ) {
    return $this->validate_required_fields( $value, array( 'accountID', 'clientID' ) );
}
```

or as suggested, `Module_Settings` could have this (`get_required_fields` could be called by `validate_required_fields`):

```php
public function validate( $value ) {
    return $this->validate_required_fields( $value );
}
```

Reply:
I think that makes sense. I was thinking to keep this along the same lines like our sanitize behavior, regardless of WordPress not having a default mechanism for setting validation ([it's been a while...](https://core.trac.wordpress.org/ticket/43208)). But overall yes, getting rid of the callback sounds good to me.

We could theoretically even do the same with sanitization to simplify: Having just a `sanitize( $value )` method and pass that directly to `sanitize_callback`.